src/models/user.ts

```diff
@@ -1,55 +1,67 @@
-import mongoose, { model, Schema } from 'mongoose'
-import bcrypt from "bcryptjs"
+import bcrypt from "bcryptjs";
+import mongoose, { model, Schema } from "mongoose";
+
+mongoose.set("toJSON", {
+  virtuals: true,
+  versionKey: false,
+  transform: function (doc, ret) {
+    delete ret._id;
+  },
+});
 
 const userSchema = new Schema({
-    email: {
-        type: String,
-        required: true,
-        unique: true,
-    },
-    password: {
-        type: String,
-        required: true,
-    },
-    role: {
-        type: String,
-        default: 'user',
-    },
-    profile: {
-        type: mongoose.Types.ObjectId,
-        ref: 'Profile',
-    },
-})
-
-userSchema.methods.checkPass = async function (password:string) {
-    const pass = await bcrypt.compare(password, this.password)
-    return pass
-}
-
-
- const User = model("User", userSchema)
-
-const Profile = mongoose.model(
-    'Profile',
-    new Schema({
-        firstName: {
-            type: String,
-            required: true,
-        },
-        lastName: {
-            type: String,
-            required: true,
-        },
-        address: {
-            type: String,
-        },
-        user: {
-            type: mongoose.Types.ObjectId,
-            ref: 'User',
-            required: true,
-            unique: true,
-        },
-    })
-)
-
-export { User, Profile }
+  email: {
+    type: String,
+    required: true,
+    unique: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  role: {
+    type: String,
+    default: "user",
+  },
+  profile: {
+    type: mongoose.Types.ObjectId,
+    ref: "Profile",
+  },
+});
+
+userSchema.methods.checkPass = async function (password: string) {
+  const pass = await bcrypt.compare(password, this.password);
+  return pass;
+};
+
+const profileSchema = new Schema({
+  firstName: {
+    type: String,
+    required: true,
+  },
+  lastName: {
+    type: String,
+    required: true,
+  },
+  address: {
+    type: String,
+  },
+  user: {
+    type: mongoose.Types.ObjectId,
+    ref: "User",
+    required: true,
+    unique: true,
+  },
+});
+
+profileSchema.virtual("name").get(function () {
+  if (!this.firstName) return "Missing name";
+  return this.firstName + " " + this?.lastName || "";
```

Comment:
Does `this?' mean it's optional?

Reply:
When we create a user, we don't create a profile for them directly since they don't provide the firstName and lastName in the same mutation.

This was picked over asking for the information since the admins will be responsible to generate the accounts of the trainees and requesting the informations directly would mean a lot of work on their side while users have the ability to edit their own profile. 

I have changed the field to reflect the models